### DISCUSSION TITLE
something wrong while I config traffic-split using admin-api

### DISCUSSION THREAD

**Discussion by Mar1usSo** [apache/apisix#6438] [Q&A]
Hello everyone,
I got error while I config traffic-split using admin-api.

Seems admin-api can not resolve "!" correctly in my reqeust.

_{
    "error_msg": "failed to check the configuration of plugin traffic-split err: failed to validate the 'vars' expression: invalid operator '!'"
}_

Here is my postman setting:
_URL:http://127.0.0.1:9080/apisix/admin/routes/395163938729231314_
Body:
_{
  "id": "395163938729231314",
  "name": "traffic-split-test",
  "methods": [
    "GET",
    "POST",
    "PUT",
    "DELETE",
    "PATCH",
    "HEAD",
    "OPTIONS",
    "CONNECT",
    "TRACE"
  ],
  "plugins": {
    "proxy-rewrite": {
      "host": "proxy.test.com"
    },
    "traffic-split": {
      "disable": false,
      "rules": [
        {
          "match": [
            {
              "vars": [
                [
                  "cookie_user_id",
                  "!",
                  "1"
                ]
              ]
            }
          ],
          "weighted_upstreams": [
            {
              "upstream_id": "368682855116571384",
              "weight": 10
            }
          ]
        }
      ]
    }
  },
  "uri": "/*",
  "hosts": [
    "demo.test.com"
  ],
  "vars": [
    [
      "http_sssss",
      "IN",
      [
        "asdfadsf"
      ]
    ],
    [
      "http_X-APP-ID",
      "IN",
      [
        "872cc0ff763db3950176427d11fa0222",
        "I872cc172771dafae017729e71f3b0059",
        "872cc172765bf5090176607739b00024",
        "2798807be64a46a8aa3f8883e48e27b4",
        "40ef0ca977344fec9b168b17ea242bd6",
        "6a470ab1696c4390b8863542c93ee723",
        "37135e7b41594fdba3057dcf63971b3f",
        "7f6b62ead3b0428da94efb379cccdf82",
        "2bac6bdbda1c494b95d4cbf26def50cd"
      ]
    ]
  ],
  "status": 1,
  "upstream_id": "368682855116571384"
}_
Result:
_{
    "error_msg": "failed to check the configuration of plugin traffic-split err: failed to validate the 'vars' expression: invalid operator '!'"
}_

Would you please help to check if sonething wrong in my request?

Thanks

**Comment by tokers** [comment]
You should have an operator after `!`, see https://github.com/api7/lua-resty-expr.

**Reply by Mar1usSo** [reply]
You are right.
But There is a operation in apisix-dashboard:
 'page.route.reverse': 'Reverse the result(!)',see https://github.com/apache/apisix-dashboard/blob/6b86ff4c4ee7b129cfeda48108fe5d4f5a2e7c6c/web/src/pages/Route/locales/en-US.ts

**Reply by Mar1usSo** [reply]
Maybe dash-board should not used "!" separated
Thanks & close this question
